rest/route/stats.go

```diff
@@ -484,3 +487,51 @@ func (tsh *taskStatsHandler) Run(ctx context.Context) gimlet.Responder {
 func makeGetProjectTaskStats(sc data.Connector) gimlet.RouteHandler {
 	return &taskStatsHandler{sc: sc}
 }
+
+type cedarTestStatsMiddleware struct{}
+
+func CheckCedarTestStats() gimlet.Middleware {
+	return &cedarTestStatsMiddleware{}
+}
+
+func (m *cedarTestStatsMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	ctx := r.Context()
+
+	newURL := fmt.Sprintf(
+		"https://cedar.mongodb.com/rest/v1/historical_test_data/%s?%s",
+		gimlet.GetVars(r)["project_id"],
+		r.URL.RawQuery,
+	)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, newURL, nil)
+	if err != nil {
+		gimlet.WriteResponse(rw, gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "problem creating cedar test stats request")))
+		return
+	}
+
+	c := utility.GetHTTPClient()
+	defer utility.PutHTTPClient(c)
+
+	resp, err := c.Do(req)
+	if err != nil {
+		gimlet.WriteResponse(rw, gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "problem sending test stats request to cedar")))
+	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+	if resp.StatusCode == http.StatusOK {
+		rw.WriteHeader(http.StatusOK)
```

Comment:
Same question about multiple responses - this and L530 could write both ok and internal server error. Is this valid?

Reply:
Yes, L530 would overwrite the status.

Reply:
The [docs](https://golang.org/pkg/net/http/#ResponseWriter) say that WriteHeader can only write a single header and other ones will be ignored and also, calling `Write()` will trigger the ok header to be written. Doesn't this mean that this if block will always return OK regardless of whether `io.Copy` succeeds?

I also tested this is a dummy program and it seems like WriteHeader only respects the first call to it - subsequent calls to WriteHeader don't change the header, so it'll return 200 ok even in the failure case.